### ISSUE TITLE
Emit a warning when iterating through a dict without calling `.items`

### ISSUE THREAD

**Issue by yory8** [PyCQA/pylint#2761]
### Is your feature request related to a problem? Please describe

I'm always forgetting to call .items() on dicts when doing stuff like:

```py
for k, v in dict.items():
    print(k, v)
```

This is a serious bug because it's only caught at runtime, and only when executing that part of the code.

### Describe the solution you'd like

I'd like it if pylint could warn me about "iterating-dict-without-items", or something like that.

**Comment by PCManticore** [comment]
Hey @yory8 Thanks for reporting the issue! This makes sense and could be a nice little new check.

**Comment by yory8** [comment]
Is there a tutorial on how to create new checkers? I'd like to give it a try

**Comment by yory8** [comment]
@PCManticore I'll give it a try myself

**Comment by yory8** [comment]
@PCManticore  I managed to write a [checker](https://gist.github.com/yory8/a4fbaa7efeb4a2730ffe16aaa38a85d2) and its [test](https://gist.github.com/yory8/aad096b308abaea16338eaf26033ac8b).

I need help with this:
1. what error code we give it? what definition?

    ```py
     msgs = {
        "W0001": (  # fixme: change error code!
            "Unpacks dict through items()",  # fixme: better definition
            "dict-missing-items",
            "Unpacking key, value of dict without calling items()",  # fixme: better definition
        )
   }
   ```
2. how to integrate it in pylint
3. I wasn't able to find how ensure that "d" in "d.items()" is a dict, so if there's some other construct that iterates through a method and gives back a tuple, it will give a false positive

**Comment by PCManticore** [comment]
Hey @yory8 Thanks for working on this!
We can probably address most of these questions once you open a PR, as it would be easier to add a feedback:
   - this check can probably live in `pylint.checkers.base` or `pylint.checkers.typecheck`, as those files are mostly related to checks that can result in an error at runtime
   - for the error code, you can probably grep the codebase for a number that's not already taken. Sorry, this is right now not very obvious, you'd have to look in the file where the check will live to figure out which number is not taken. For instance, the latest warning from `pylint.checkers.base` is `W0143` so in your case you'd add yours as `W0144`
    - also make sure to read http://pylint.pycqa.org/en/latest/development_guide/contribute.html in case there's something useful for you there
    - Regarding question 3, you can probably use inference instead http://pylint.pycqa.org/projects/astroid/en/latest/inference.html. Check `pylint.checkers.utils.safe_infer` It's a function that receives a node and tries to `infer` what the runtime value of that node would be, if executed. In your case, if you pass a variable `d` to it, it would say, if it can infer the value, that it is a dictionary, in which case you'd know to emit the message you just implemented.

Let me know if this answers your questions. Once we have a PR against `pylint`, it's going to be easier to leave more feedback.

**Comment by yory8** [comment]
@PCManticore yes, thank'you! this should be enough info. I'll work on this soon
